src/playlist-loader.js

```diff
@@ -379,57 +389,104 @@
       return 0;
     }
 
-    // 1) Walk backward until we find the latest segment with timeline
+    // 1) Walk backward until we find the first segment with timeline
     // information that is earlier than `time`
-    targetDuration = this.media_.targetDuration || 10;
-    i = this.media_.segments.length;
-    while (i--) {
+    for (i = lastSegment; i > 0; i--) {
       segment = this.media_.segments[i];
       if (segment.end !== undefined && segment.end <= time) {
-        time -= segment.end;
+        startIndex = i + 1;
+        knownStart = segment.end;
+        if (startIndex >= numSegments) {
+          // Somehow the last segment claims to end *before* the time we are
+          // searching for so just return it
```

Comment:
Ok, that makes more sense. Can you replace "somehow" in the comment with that example?